__tests__/posts.ts

```diff
@@ -2138,3 +2141,207 @@ describe('mutation checkLinkPreview', () => {
+      },
+    ]);
+  });
+
+  it('should not authorize when not logged in', () =>
+    testMutationErrorCode(
+      client,
+      { mutation: MUTATION, variables: params },
+      'UNAUTHENTICATED',
+    ));
+
+  it('should return an error if post type is not allowed to be editable', async () => {
+    loggedUser = '1';
+    await con
+      .getRepository(Post)
+      .update({ id: 'p1' }, { type: PostType.Share });
+
+    await testMutationErrorCode(
+      client,
+      { mutation: MUTATION, variables: params },
+      'FORBIDDEN',
+    );
+
+    await con
+      .getRepository(Post)
+      .update({ id: 'p1' }, { type: PostType.Article });
+
+    return testMutationErrorCode(
+      client,
+      { mutation: MUTATION, variables: params },
+      'FORBIDDEN',
+    );
+  });
+
+  it('should return an error if title exceeds 80 characters', async () => {
+    loggedUser = '1';
+
+    return testMutationErrorCode(
+      client,
+      {
+        mutation: MUTATION,
+        variables: {
+          ...params,
+          title:
+            'Hello World! Start your squad journey here - Hello World! Start your squad journey here',
+        },
+      },
+      'GRAPHQL_VALIDATION_FAILED',
+    );
+  });
+
+  it('should return an error if content exceeds 4000 characters', async () => {
+    loggedUser = '1';
+
+    const content = 'Hello World! Start your squad journey here';
+    const sample = new Array(100).fill(content);
+
+    return testMutationErrorCode(
+      client,
+      {
+        mutation: MUTATION,
+        variables: {
+          ...params,
+          content: sample.join(' - '),
+        },
+      },
+      'GRAPHQL_VALIDATION_FAILED',
+    );
+  });
+
+  it('should restrict member when user is not the author of the post', async () => {
+    loggedUser = '2';
+
+    return testMutationErrorCode(
+      client,
+      { mutation: MUTATION, variables: { id: 'p1', title: 'Test' } },
+      'FORBIDDEN',
+    );
+  });
+
+  it('should not update title if the post type is "welcome"', async () => {
+    loggedUser = '1';
+    const title = 'Updated title';
+    const res = await client.mutate(MUTATION, {
+      variables: { id: 'p1', title },
+    });
+    expect(res.errors).toBeFalsy();
+    const actual = await con.getRepository(Post).findOneBy({ id: 'p1' });
+    expect(actual.title).not.toEqual(title);
+  });
+
+  it('should update title if the post type is "freeform"', async () => {
+    loggedUser = '1';
+    await con
+      .getRepository(Post)
+      .update({ id: 'p1' }, { type: PostType.Freeform });
+    const title = 'Updated title';
+    const res = await client.mutate(MUTATION, {
+      variables: { id: 'p1', title },
+    });
+    expect(res.errors).toBeFalsy();
+    const actual = await con.getRepository(Post).findOneBy({ id: 'p1' });
+    expect(actual.title).toEqual(title);
+  });
+
+  it('should allow moderator to do update even when user is not the author of the post', async () => {
```

Comment:
Are you sure mods and admins can edit my post?

Reply:
From the moderation spec, it did says it should. Let me confirm with Sab regarding freeform posts.